go/vt/discovery/healthcheck.go

```diff
@@ -548,41 +616,23 @@ func (hcc *healthCheckConn) setServingState(serving bool, reason string) {
 
 // stream streams healthcheck responses to callback.
 func (hcc *healthCheckConn) stream(ctx context.Context, hc *HealthCheckImpl, callback func(*querypb.StreamHealthResponse) error) {
-	hcc.mu.Lock()
-	conn := hcc.conn
-	hcc.mu.Unlock()
-
-	if conn == nil {
-		var err error
-		conn, err = tabletconn.GetDialer()(hcc.tabletStats.Tablet, grpcclient.FailFast(true))
+	if hcc.conn == nil {
```

Comment:
I just tried making this change, but I encountered a readability problem: `checkConn` calls `hc.connsWG.Done()`, which belongs to `HealthCheckImpl`. This would be confusing if we moved this to a different file. Also, `checkConn` references multiple other things in `hc`. So, we'll probably need to do a deeper refactor to clean up this mess.

But returns are diminishing.